Making profile form more accessible

singingwolfboy opened the pull request:
This pull request does the following things, all related to the profile form:

* Replace `<div class="section-header">` elements with `<h3>` elements
* Add accessible `<label>` elements to the date fields and the select fields
* Ensure that every select field has a unique ID, so that we can use `aria-describedby` attributes to label error messages in a future pull request
* Pull Sass related to the profile form out into its own `profile-form.scss` file, for organizational purposes

To test this pull request, I suggest using [ChromeVox](http://www.chromevox.com/) or [VoiceOver](https://help.apple.com/voiceover/info/guide/10.12/). To activate/deactivate VoiceOver on a Mac, press Command + F5. Different browsers & screen readers work differently; for example, when reading the "Date of birth" fields, both ChromeVox and VoiceOver with Chrome read out "Day, edit text", "Month, edit text", "Year, edit text". However, VoiceOver with Safari reads out "Day, edit text, date of birth", "Month, edit text, date of birth", "Year, edit text, date of birth", which is much clearer for a non-sighted user. (This is related to #1683.)

A few concrete manual test cases:

* For every select field, try clicking the label, and ensure that the browser automatically focuses on the select field.
* Turn on ChromeVox or VoiceOver and tab through the profile form. All fields should announce themselves properly, with the following known exceptions:
  * the "Gender" radio fields are not properly labelled
  * error states, such as not filling in required fields, are not properly accessible

![screen shot 2016-11-17 at 11 49 12 am](https://cloud.githubusercontent.com/assets/132355/20398579/f085cdb6-acbb-11e6-8c20-ad214238ffe2.png)
![screen shot 2016-11-17 at 11 48 32 am](https://cloud.githubusercontent.com/assets/132355/20398584/f7e858b2-acbb-11e6-982a-75471d1b31fa.png)
![screen shot 2016-11-17 at 11 48 37 am](https://cloud.githubusercontent.com/assets/132355/20398589/fbe3d90a-acbb-11e6-847a-91ae49fc1b23.png)

pdpinch commented:
Maybe this is moot because this is WIP, but I have no idea what the scope of this is. Does it address a particular issue?

noisecapella commented:
Can you rebase please?

noisecapella commented on static/js/util/profile_edit_test.js:
```diff
@@ -145,10 +145,11 @@ describe('Profile Editing utility functions', () => {
+      let legend, space;  // eslint-disable-line no-unused-vars
       if (omitDay) {
-        [monthTextField, , yearTextField] = renderDateField(true);
+        [legend, space, monthTextField, , yearTextField] = renderDateField(true);
```
As long as you're here, do you mind refactoring this function to use enzyme, if you feel like it? `renderDateField` is kind of a hacky workaround at the moment

noisecapella replied:
Made #1774 for this

noisecapella commented:
It looks like the next button on the signup page is now causing a full page refresh. Can you investigate a fix?

noisecapella commented:
There is a change of appearance in the user page when I edit the personal information. Before:
![screenshot from 2016-11-15 15-42-07](https://cloud.githubusercontent.com/assets/863262/20323311/247969e8-ab4a-11e6-97d1-03c66598141f.png)

After:
![screenshot from 2016-11-15 15-41-34](https://cloud.githubusercontent.com/assets/863262/20323317/28f46662-ab4a-11e6-8272-4fd18fd79f14.png)

singingwolfboy commented:
@noisecapella Thanks for catching that. The page refresh was due to me swapping a `<section>` element with a `<form>` element -- looks like introducing true forms is going to be tricker than I expected, and will almost certainly require using the [redux-form](http://redux-form.com/) library, so I've reverted that change. I've also fixed the styling problems with the profile edit form, although there may still be some fiddly form label issues that might need some attention from @roberthouse54 at some point.

noisecapella commented:
It seems like there's still a regression in the style, though it's smaller than before:

![screenshot from 2016-11-15 16-40-54](https://cloud.githubusercontent.com/assets/863262/20325372/5a694b6a-ab52-11e6-8b49-250fde91bf11.png)

Note that the text and select items aren't aligned anymore like in the 'before' image above

noisecapella commented:
The education form is also affected:
![screenshot from 2016-11-15 16-42-44](https://cloud.githubusercontent.com/assets/863262/20325420/93628454-ab52-11e6-81f3-f8ae64d36d32.png)

noisecapella commented:
Please add the 'Needs Review' label back when you are done addressing comments

singingwolfboy commented:
I spent some time with @roberthouse54 going over the styling in this pull request, and we made some changes to improve it. It's still not pixel-perfect, but it looks much more reasonable now, and nothing seems obviously broken. I've updated the screenshots in the description of this pull request to reflect the way the pull request currently looks.

I'll be out all of next week, so if someone else wants to pick this up while I'm gone, that's fine with me. Otherwise, we can either merge it before I leave for the day, and clean up the little issues in a future pull request; or we can hold off until I get back.

noisecapella commented:
Looks like the style issues are fixed. I'll proceed with the manual steps in your description

noisecapella commented:
Looks good, I'll merge this for you :+1:
